raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1306,26 +1297,33 @@ public boolean isRunning() {
         return gracefulShutdown == null || !gracefulShutdown.isFinished();
     }
 
+    public boolean isShuttingDown() {
+        GracefulShutdown gracefulShutdown = shutdown.get();
+        return gracefulShutdown != null && !gracefulShutdown.isFinished();
+    }
+
     private void pollShutdown(GracefulShutdown shutdown) throws IOException {
         // Graceful shutdown allows a leader or candidate to resign its leadership without
-        // awaiting expiration of the election timeout. During shutdown, we no longer update
-        // quorum state. All we do is check for epoch updates and try to send EndQuorumEpoch request
-        // to finish our term. We consider the term finished if we are a follower or if one of
-        // the remaining voters bumps the existing epoch.
+        // awaiting expiration of the election timeout. As soon as another leader is elected,
```

Comment:
I'm slightly prefer the former comment which is more clear about what we are expecting.

Reply:
What specifically? The previous comment was no longer accurate, so I changed it.